app/src/core/components/Modal.tsx

```diff
@@ -0,0 +1,31 @@
+import Dialog, { DialogProps } from '@mui/material/Dialog';
+import DialogContent from '@mui/material/DialogContent';
+import useMediaQuery from '@mui/material/useMediaQuery';
+import { useTheme } from '@mui/material/styles';
+import { DialogActions } from '@mui/material';
+
+interface ResponsiveModalProps extends DialogProps {
+    onCloseCallback?: () => void;
+    closeDisabled?: boolean;
+    actionChildren?: DialogProps['children'];
+}
+
+export default function Modal({ children, onCloseCallback, closeDisabled, actionChildren, ...rest }: ResponsiveModalProps) {
+    const theme = useTheme();
+    const fullScreen = useMediaQuery(theme.breakpoints.down('md'));
```

Comment:
an optional `fullScreen` prop on `Modal` would be nice to allow overriding this behavior. as i can see some occasions where we'd want full screen on large screens, or not want full screen on smaller screens.